--- glibc-2.24/nss/nsswitch.c.nixpath	2016-12-02 23:11:20.528355442 +0100
+++ glibc-2.24/nss/nsswitch.c	2016-12-03 00:00:05.237349358 +0100
@@ -34,6 +34,7 @@
 #include <shadow.h>
 
 #if !defined DO_STATIC_NSS || defined SHARED
+# include <unistd.h>   /* for __libc_enable_secure */
 # include <gnu/lib-names.h>
 #endif
 
@@ -320,6 +321,16 @@
 				*(const char *const *) p2);
 }
 
+static const char *const nss_prefixes[] = { "/run/nss-modules-1/", "/run/nss-modules-2/" };
+
+void *
+nss_load_library_prefix (const char* prefix, char *name)
+{
+  size_t shlen = (strlen (prefix) + strlen (name) + 1);
+  char shlib_name[shlen];
+  __stpcpy (__stpcpy (shlib_name, prefix), name);
+  return __libc_dlopen (shlib_name);
+}
 
 #if !defined DO_STATIC_NSS || defined SHARED
 /* Load library.  */
@@ -347,6 +358,8 @@
 		      + strlen (__nss_shlib_revision) + 1);
       int saved_errno = errno;
       char shlib_name[shlen];
+      int i;
+      void *lib_handle = NULL;
 
       /* Construct shared object name.  */
       __stpcpy (__stpcpy (__stpcpy (__stpcpy (shlib_name,
@@ -355,7 +368,13 @@
 			  ".so"),
 		__nss_shlib_revision);
 
-      ni->library->lib_handle = __libc_dlopen (shlib_name);
+      if (__builtin_expect (! __libc_enable_secure, 1))
+        {
+          for (i = 0; lib_handle == NULL && i < sizeof (nss_prefixes); i++)
+            lib_handle = nss_load_library_prefix (nss_prefixes[i], shlib_name);
+        }
+
+      ni->library->lib_handle = lib_handle ?: __libc_dlopen (shlib_name);
       if (ni->library->lib_handle == NULL)
 	{
 	  /* Failed to load the library.  */
