diff -ur glibc-2.24-orig/nss/nsswitch.c glibc-2.24/nss/nsswitch.c
--- glibc-2.24-orig/nss/nsswitch.c	2016-12-07 13:21:53.630202041 +0100
+++ glibc-2.24/nss/nsswitch.c	2016-12-14 16:01:19.595104801 +0100
@@ -320,6 +320,24 @@
 				*(const char *const *) p2);
 }
 
+#ifndef UNIQUE_ID
+# error UNIQUE_ID is undefined
+#endif
+
+static const char *nss_root = "/run/nss-modules/" UNIQUE_ID "/";
+
+void *
+nss_load_library_prefix (const char *name)
+{
+  size_t shlen = (strlen (nss_root) + strlen (name) + 1);
+  char shlib_name[shlen];
+
+  __stpcpy (__stpcpy (shlib_name,
+                      nss_root),
+            name);
+  
+  return __libc_dlopen (shlib_name);
+}
 
 #if !defined DO_STATIC_NSS || defined SHARED
 /* Load library.  */
@@ -347,7 +365,8 @@
 		      + strlen (__nss_shlib_revision) + 1);
       int saved_errno = errno;
       char shlib_name[shlen];
-
+      void *lib_handle = NULL;
+      
       /* Construct shared object name.  */
       __stpcpy (__stpcpy (__stpcpy (__stpcpy (shlib_name,
 					      "libnss_"),
@@ -355,7 +374,10 @@
 			  ".so"),
 		__nss_shlib_revision);
 
-      ni->library->lib_handle = __libc_dlopen (shlib_name);
+      if (lib_handle == NULL)
+        lib_handle = nss_load_library_prefix ((const char *) shlib_name);
+
+      ni->library->lib_handle = lib_handle ?: __libc_dlopen (shlib_name);
       if (ni->library->lib_handle == NULL)
 	{
 	  /* Failed to load the library.  */
